contrib/build-push-cephonly-base-images.sh

```diff
@@ -0,0 +1,157 @@
+
+# Build and push daemon-base images which are tagged with the version of Ceph installed in them.
+# Images are intended for Rook and are therefore CentOS-based.
+#
+# This script will look to see which Ceph package versions are available for given Ceph named
+# versions (e.g., luminous, mimic). If any of the available Ceph package versions don't have a
+# corresponding image and tag on the registry, the script will build the image and push it to the
+# registry. The script will do nothing if images are at parity with packages.
+#
+# This script can therefore be run on a cron regularly, and very little manual maintenance should be
+# necessary unless Ceph RPM releases or images published change. The only maintenance needed is to
+# add new Ceph releases to the `FLAVORS_TO_BUILD` variable below.
+
+
+#---------------------------------------------------------------------------------------------------
+# YOU CAN TUNE A SCRIPT, BUT YOU CAN'T TUNE A FISH
+#---------------------------------------------------------------------------------------------------
+
+# This is mostly an internal representation of `FLAVORS_TO_BUILD`. This script builds x86 and arm
+# images for centos, so the middle item doesn't really matter except to make this easier to grok at
+# first glance.
+#   Usage examples:
+# luminous,centos,7 = build luminous containers on centos 7
+# mimic,centos,7 = build mimic containers on centos 7
+# nautilus,centos,8 = build nautilus containers on centos 8
+FLAVORS_TO_BUILD="luminous,centos,7 mimic,centos,7"
+
+# Use the ceph library by default
+PUSH_LIBRARY="ceph"
+
+# Instead of 'daemon-base', call these images 'ceph' so users don't get these confused with the
+# daemon-base and daemon images
+PUSH_IMAGE_NAME="ceph"
+
+# Push to the docker hub by default
+PUSH_REGISTRY_URL="https://registry.hub.docker.com/v2/repositories/${PUSH_LIBRARY}/${PUSH_IMAGE_NAME}"
+
+
+#---------------------------------------------------------------------------------------------------
+# FUNCTIONS
+#---------------------------------------------------------------------------------------------------
+
+# Return a list of the ceph version strings available on the server as:
+#   "12.0.0 12.0.1 12.0.2 12.1.0 ..."
+function ceph_versions_on_server() {
+  ceph_codename="${1}" ; distro_release="${2}" ; ceph_arch="${3}"
+  download_url="http://download.ceph.com/rpm-${ceph_codename}/el${distro_release}/${ceph_arch}/"
+  # Curl gives a listing of the URL like below:
+  # <a href="ceph-12.2.7-0.el7.x86_64.rpm">ceph-12.2.7-0.el7.x86_64.rpm</a>  17-Jul-2018 14:11  3024
+  # The ceph base package can be id'ed uniquely by the text ">ceph-" followed by a version string
+  pkg_regex=">ceph-[0-9.]+"
+  # pkg_list is returned in the form ">ceph-12.0.0 >ceph-12.0.1 >ceph-12.0.2 ..."
+  pkg_list="$(curl --silent "${download_url}" | grep --extended-regexp --only-matching "${pkg_regex}")"
```

Comment:
This feels similar to this wip code: https://github.com/ceph/ceph-container/pull/1108/files
Perhaps it can help

Reply:
@leseb @travisn @liewegas  I've been thinking more about this comment and my implementation in general. I think there is a potential issue with both my and with the #1108 implementation which is that there are not always versions of ceph for both x86_64 and aarch64. By looking, it appears that sometimes arm builds aren't done every time an x86_64 build is done. If we want to have a tag like `ceph/ceph:version` that will automatically get x86_64 or arm versions, we need to have both versions available.

For luminous, x86_64 versions  have 12.0.{0,1,2} which arm doesn't have builds for. aarch64 images start with 12.0.3. I'm assuming that this is a situation that could occur in the future. I think we can (and should) build images for both architectures, but we should not build a base package for it. Like we can have images `ceph/ceph-x86_64:12.0.2` without having a corresponding arm image, but we should not build a ceph image until we have both versions; so in my example below, the first ceph version would be `ceph/ceph:12.0.3` when we finally have an arm build. I think by doing this, we can still build the latest, but the Rook project will not need to do extra work to make sure both x86_64 and arm images exist since the existence of a `ceph` image implies that both exist.
```
ceph-12.0.0-0.el7.x86_64.rpm                       08-Feb-2017 16:40                3060
ceph-12.0.1-0.el7.x86_64.rpm                       27-Mar-2017 18:05                3060
ceph-12.0.2-0.el7.x86_64.rpm                       24-Apr-2017 14:06                3060
ceph-12.0.3-0.el7.x86_64.rpm                       17-May-2017 21:11                3060
ceph-12.1.0-0.el7.x86_64.rpm                       23-Jun-2017 13:42                3024
...
```
```
ceph-12.0.3-0.el7.aarch64.rpm                      23-Jun-2017 13:42                3032
ceph-12.1.0-0.el7.aarch64.rpm                      23-Jun-2017 13:46                2996
...
```